### PR TITLE
Changed tags for legacy flannel images

### PR DESCRIPTION
**Pull request by atsikham**

### CODE REVIEW

**Comment by przemyslavic** [comment]
Now the situation is somewhat the opposite.
It works for `epicli apply` (dev) and `epicli upgrade` (dev to dev) but fails for `epicli upgrade` 0.9 to dev.

```
13:57:00 INFO cli.engine.ansible.AnsibleCommand - TASK [Download image file] *****************************************************
13:57:00 INFO cli.engine.ansible.AnsibleCommand -
13:57:00 INFO cli.engine.ansible.AnsibleCommand - TASK [download : Download image flannel-v0.11.0-amd64.tar] *********************
13:57:01 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-xx-xx-xx-xx.eu-west-1.compute.amazonaws.com]
13:57:01 INFO cli.engine.ansible.AnsibleCommand -
13:57:01 INFO cli.engine.ansible.AnsibleCommand - TASK [image_registry : Load image epiphanyplatform/flannel:v0.11.0-amd64] ******
13:57:04 INFO cli.engine.ansible.AnsibleCommand - changed: [ec2-xx-xx-xx-xx.eu-west-1.compute.amazonaws.com]
13:57:04 INFO cli.engine.ansible.AnsibleCommand -
13:57:04 INFO cli.engine.ansible.AnsibleCommand - TASK [image_registry : Tag image epiphanyplatform/flannel:v0.11.0-amd64 with ec2-xx-xx-xx-xx.eu-west-1.compute.amazonaws.com:5000/epiphanyplatform/flannel:v0.11.0-amd64] ***
13:57:05 ERROR cli.engine.ansible.AnsibleCommand - fatal: [ec2-xx-xx-xx-xx.eu-west-1.compute.amazonaws.com]: FAILED! => {"changed": true, "cmd": "docker tag epiphanyplatform/flannel:v0.11.0-amd64 ec2-xx-xx-xx-xx.eu-west-1.compute.amazonaws.com:5000/epiphanyplatform/flannel:v0.11.0-amd64", "delta": "0:00:00.050270", "end": "2021-03-05 13:57:05.316807", "msg": "non-zero return code", "rc": 1, "start": "2021-03-05 13:57:05.266537", "stderr": "Error response from daemon: No such image: epiphanyplatform/flannel:v0.11.0-amd64", "stderr_lines": ["Error response from daemon: No such image: epiphanyplatform/flannel:v0.11.0-amd64"], "stdout": "", "stdout_lines": []}

```

**Comment by przemyslavic** [comment]
Tested:
`epicli apply` (dev)
`epicli apply` (0.9)
`epicli upgrade` (0.9 to dev)
`epicli upgrade` (dev to dev)
on Azure Ubuntu with no issues.
```
[operations@qa-09todev-repository-vm-0 ~]$ sudo docker images | grep -i flannel
quay.io/coreos/flannel                                               v0.12.0               4e9f801d2217        12 months ago       52.8MB
quay.io/coreos/flannel                                               v0.12.0-amd64         4e9f801d2217        12 months ago       52.8MB
qa-09todev-repository-vm-0:5000/quay.io/coreos/flannel               v0.12.0               4e9f801d2217        12 months ago       52.8MB
qa-09todev-repository-vm-0:5000/quay.io/coreos/flannel               v0.12.0-amd64         4e9f801d2217        12 months ago       52.8MB
epiphanyplatform/flannel                                             v0.11.0               ff281650a721        2 years ago         52.6MB
epiphanyplatform/flannel                                             v0.11.0-amd64         ff281650a721        2 years ago         52.6MB
qa-09todev-repository-vm-0:5000/epiphanyplatform/flannel             v0.11.0               ff281650a721        2 years ago         52.6MB
qa-09todev-repository-vm-0:5000/epiphanyplatform/flannel             v0.11.0-amd64         ff281650a721        2 years ago         52.6MB
quay.io/coreos/flannel                                               v0.11.0               ff281650a721        2 years ago         52.6MB
quay.io/coreos/flannel                                               v0.11.0-amd64         ff281650a721        2 years ago         52.6MB
```
```
[operations@qa-devtodev-repository-vm-0 ~]$ sudo docker images | grep -i flannel
quay.io/coreos/flannel                                                v0.12.0               4e9f801d2217        12 months ago       52.8MB
quay.io/coreos/flannel                                                v0.12.0-amd64         4e9f801d2217        12 months ago       52.8MB
qa-devtodev-repository-vm-0:5000/quay.io/coreos/flannel               v0.12.0               4e9f801d2217        12 months ago       52.8MB
qa-devtodev-repository-vm-0:5000/quay.io/coreos/flannel               v0.12.0-amd64         4e9f801d2217        12 months ago       52.8MB
epiphanyplatform/flannel                                              v0.11.0               ff281650a721        2 years ago         52.6MB
epiphanyplatform/flannel                                              v0.11.0-amd64         ff281650a721        2 years ago         52.6MB
qa-devtodev-repository-vm-0:5000/epiphanyplatform/flannel             v0.11.0               ff281650a721        2 years ago         52.6MB
qa-devtodev-repository-vm-0:5000/epiphanyplatform/flannel             v0.11.0-amd64         ff281650a721        2 years ago         52.6MB
```

**Review by seriva — DISMISSED** [review]

File: core/src/epicli/data/common/ansible/playbooks/roles/image_registry/tasks/main.yml [hunk 1]
```diff
@@ -42,34 +42,53 @@
...
+          include_tasks: load-image.yml
+          loop: "{{ specification.images_to_load.legacy }}"
+
+        # installations before v0.10.0 have no "epiphanyplatform" flannel images
+        # that were changed due to replacement Skopeo with Crane
+        # following task can be considered for removal after the end of v0.6 support
+        - name: Tag legacy flannel images
+          vars:
+            docker_image: "{{ item }}"
+          include_tasks: tag-flannel-images.yml
```
Comment (mkyc) [inline core/src/epicli/data/common/ansible/playbooks/roles/image_registry/tasks/main.yml:85]: shouldn't this task be named `tag-quay-io-coreos-images.yml`? Or make it bit more general and make it even `retag-images.yml` ?

**Review by mkyc — COMMENTED** [review]
One question or suggestion from me.

File: core/src/epicli/data/common/ansible/playbooks/roles/image_registry/tasks/main.yml [hunk 2]
```diff
@@ -42,34 +42,53 @@
         docker_image: "{{ specification.registry_image }}"
 
     - name: Check if registry is running
-      become: yes
-      shell: docker ps | grep registry:2 | cat
+      become: true
+      command: docker ps -q -f 'ancestor=registry:2' -f 'status=running'
```
Comment (to-bar) [inline core/src/epicli/data/common/ansible/playbooks/roles/image_registry/tasks/main.yml:46]: :star:

File: core/src/epicli/data/common/ansible/playbooks/roles/image_registry/tasks/main.yml (hunk 1 above)
Comment (atsikham) [reply core/src/epicli/data/common/ansible/playbooks/roles/image_registry/tasks/main.yml:85]: There are other "quay.io/coreos" images that were not moved to "epiphanyplatform". This is only related to old flannel v0.11 images and can be removed after the end of Epiphany v0.6 support.
Can we leave it as is in this case?

**Review by mkyc — DISMISSED** [review]

**Review by seriva — APPROVED** [review]

**Review by mkyc — APPROVED** [review]

**Comment by atsikham** [comment]
Will be merged after local verification and test from @przemyslavic

**Review by to-bar — APPROVED** [review]

**Comment by przemyslavic** [comment]
Tested:
epicli apply (dev)
epicli upgrade (0.6 to dev)
epicli upgrade (0.9 to dev)
epicli upgrade (dev to dev)

@atsikham you can merge 👍

**Review by przemyslavic — APPROVED** [review]

**Pull request merged**